io/private/las/Vlr.cpp

```diff
@@ -0,0 +1,412 @@
+/******************************************************************************
+* Copyright (c) 2014, Hobu Inc. (hobu@hobu.co)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "Vlr.hpp"
+
+#include <nlohmann/json.hpp>
+#include <pdal/util/FileUtils.hpp>
+#include <pdal/util/Inserter.hpp>
+#include <pdal/util/Extractor.hpp>
+
+namespace pdal
+{
+namespace las
+{
+
+VlrList parseIgnoreVlrs(const StringList& ignored)
+{
+    std::string error;
+    return parseIgnoreVlrs(ignored, error);
+}
+
+VlrList parseIgnoreVlrs(const StringList& ignored, std::string& error)
+{
+    error.clear();
+
+    // Always ignore COPC VLRs and superseded VLRs.
+    VlrList ignoredVlrs { {"copc", 1000}, {"LASF_Spec", 7} };
```

Comment:
We should ignore ALL COPC VLRs #3645 3649

Reply:
You said that you wanted to have the info VLR in metadata so that you could know the input was a COPC file. It's fine if we don't *write* the COPC VLRs in output with LAS.

Reply:
Upon conservation of the worry of advertising files that are written with PDAL that appear to be COPC that aren't actually clustered, I think we should ignore all COPC VLRs on read.